cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tether/TetherClientHandler.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.tether;
+
+import com.google.gson.Gson;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.namespace.NamespaceAdmin;
+import io.cdap.cdap.proto.NamespaceMeta;
+import io.cdap.common.http.HttpMethod;
+import io.cdap.common.http.HttpResponse;
+import io.cdap.http.AbstractHttpHandler;
+import io.cdap.http.HttpResponder;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import javax.inject.Inject;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+
+/**
+ * {@link io.cdap.http.HttpHandler} to manage tethering client v3 REST APIs
+ */
+@Path(Constants.Gateway.API_VERSION_3)
+public class TetherClientHandler extends AbstractHttpHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(TetherClientHandler.class);
+  private static final Gson GSON = new Gson();
+  static final String CREATE_TETHER = "/v3/tethering/connect";
+  // Following constants are same as defined in KubeMasterEnvironment
+  static final String NAMESPACE_CPU_LIMIT_PROPERTY = "k8s.namespace.cpu.limits";
```

Comment:
we should not have k8s specific details in appfabric. Can we instead just use Map of properties in namespace allocation?

Reply:
The client (via UI) just specifies the names of CDAP namespaces that will be part of the tethering. Tether client backend then needs to get the corresponding k8s namespace and resource quota (if configured) from NamespaceConfig in NamespaceMeta. k8s properties are stored in NamespaceConfig, which is basically a Map<String, String>. 
Duplicating the constants here is not ideal, but we don't have a better way at the moment - can't depend on cdap-kubernetes/ in appfabric

Reply:
discussed offline, fixed.